app/controllers/user_annotations_controller.rb

```diff
@@ -8,6 +8,9 @@ class UserAnnotationsController < ApplicationController
   def index
     #get all this user's annotations
     @user_annotations = current_user.user_annotations.owned_by(current_user)
+    views = UserAnnotation.viewable(current_user)
+    edits = UserAnnotation.editable(current_user)
```

Comment:
We actually don't need to concatenate the views & edits as edits is a subgroup of views.  You can just pull all the annotations viewable by the user, and then in the index view, depending on whether the user can edit or not, render the correct buttons.